common/src/lib.rs

```diff
@@ -224,9 +224,10 @@ macro_rules! generate_setup_and_run_command {
         {
             use futures::TryStreamExt;
             let futures = futures::stream::FuturesUnordered::new();
+            let mut networks = std::collections::VecDeque::from(network);
 
             $(
-                let config = crate::$config::new(client.pop().expect("Not enough clients"), pallet_tx.clone(), network.pop().expect("Not enough networks"), logger.clone(), account_id.clone(), key_store.clone(), prometheus_config.clone()).await?;
+                let config = crate::$config::new(client.pop().expect("Not enough clients"), pallet_tx.clone(), networks.pop_front().expect("Not enough networks"), logger.clone(), account_id.clone(), key_store.clone(), prometheus_config.clone()).await?;
```

Comment:
We pop front on the networks here to ensure consistent order, however, we do not do the same for the clients.

Reply:
It does not matter for the clients, they are the same.